CONTRIBUTING.md

```diff
@@ -99,30 +140,70 @@ Note in particular the
 for how to view the diffs in the event of snapshot test failures, and also how to update the saved snapshot
 of one or all snapshot failures.
 
-## Setting up build environment
+### Test Coverage
 
-This project uses Node 14.x, if you need to use a different version, look at using `nvm` to manage your Node versions. If you are using `nvm`, you can install the 14.x version of Node with `nvm install 14.19.1; nvm use 14.19.1`.
+`yarn run jest --coverage` will generate a coverage report in the `coverage` directory, which is ignored by this project's `.gitignore`.
+Your IDE may also be able to show you the test coverage of a source file.
+Adding tests where possible - see [Location of code](#location-of-code) for constraints to code not currently covered by the automated tests is a great way to contribute!
 
-To setup the local environment after cloning the repository, run the following commands:
+## Dependency Upgrades and Respository Maintenance
 
-```shell
-yarn
-yarn build
-yarn test
-yarn lint
-```
+This section is most useful for maintainers of the repository and not necessary for typical documentation, code, or test contributions.
+
+<details><summary>Expand Details on Dependency Upgrades and Dependabot</summary>
+
+Keeping dependencies up to date ensures the best experience and security for Tasks plugin users.
+This project uses Dependabot to help automate dependency updates, but some dependencies require manual testing
+before they can be merged.
+See the [FAQ entry on smoke-testing](#how-do-i-smoke-test-the-tasks-plugin) for how to manually test.
+Multiple depdendency upgrades can be smoke-tested together in a batch.
+
+### Thought-Process for Deciding Whether a Dependency Needs Manual Testing
+
+- When in doubt, smoke-test. Smoke-testing of multiple dependency upgrades can be done in a batch, to reduce the time spent on this process.
+- **Definitely smoke test**: Anything that is involved in producing the built output to users. For example:
+  - esbuild (the build system)
+  - anything mentioned in the esbuild config file (e.g. `builtin-modules`, `svelte-preprocess`)
+  - obsidian
+  - all @codemirror
+  - moment
+  - everything in the "dependencies" (rather than dev-dependencies) list
```

Comment:
I can't see `dev-dependencies` - I'm guessing that means `devDependencies` in `package.json`...